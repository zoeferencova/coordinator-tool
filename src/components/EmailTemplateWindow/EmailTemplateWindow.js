
import React from 'react';
import AppContext from '../../contexts/contexts'
import styles from './EmailTemplateWindow.module.css'
import { style } from 'd3';

export default class EmailTemplateWindow extends React.Component {
    static contextType = AppContext;

    render() {
        return (
            <div className={style.box}>
                <div className={styles.templateContent} itemkey={this.props.id}>
                    <p className={styles.templateSubject}>{this.props.template_subject}</p>
                    <p className={styles.templateBody}>{this.props.template_content}</p>
                </div>
            </div>
            
        )
    }
}