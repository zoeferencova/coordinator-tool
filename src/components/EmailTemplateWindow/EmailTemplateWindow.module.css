

.templateBody {
    white-space: pre-line;
}

.templateSubject {
    font-weight: bold;
}

.templateContent {
    text-align: left;
    padding: 0px 30px;
    font-size: 14px;
}

.templateContent Button {
    float: right;
}

.templateItem:hover {
    cursor: pointer;
}

@media only screen and (max-width: 850px) {
    .templateContent {
        padding: 10px;
        font-size: 12px;
    }
}
