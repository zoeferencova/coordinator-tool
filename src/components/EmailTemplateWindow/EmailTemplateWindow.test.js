import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EmailTemplateWindow from './EmailTemplateWindow'

describe(`EmailTemplate component`, () => {
  it('renders a .EmailTemplateWindow by default', () => {
    const wrapper = shallow(<EmailTemplateWindow />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})