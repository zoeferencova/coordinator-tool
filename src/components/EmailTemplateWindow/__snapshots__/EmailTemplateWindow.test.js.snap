// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailTemplate component renders a .EmailTemplateWindow by default 1`] = `
<div>
  <div
    className="templateContent"
  >
    <p
      className="templateSubject"
    />
    <p
      className="templateBody"
    />
  </div>
</div>
`;
